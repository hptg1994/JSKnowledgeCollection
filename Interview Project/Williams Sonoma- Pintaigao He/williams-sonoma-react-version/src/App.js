import React from 'react';
import * as history from 'history';
import { Router, Route, Switch, Link, NavLink } from 'react-router-dom';
import MainPage from './Pages/Main';
import DetailPage from './Pages/Detail';
const histories = history.createBrowserHistory()

class App extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <Router history={histories}>
        <div>
          <Switch>
            <Route path="/" component={MainPage} exact={true} />
            <Route path="/item/:id" component={DetailPage} />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
