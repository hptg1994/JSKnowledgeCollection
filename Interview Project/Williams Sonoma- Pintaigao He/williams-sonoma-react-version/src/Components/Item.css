.item-container {
    position: relative;
    display: inline-block;
    transition: 0.5s;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
}

.item-container:hover {
    box-shadow: 0px 0px 20px lightgrey;
    transition: 0.5s;
}

.item-container-image {
    position: relative;
    display: inline-block;
}

.item-container-price {
    position: absolute;
    bottom: 20px;
    padding: 10px 20px;
    left: 0px;
    background-color: white;
}

.item-container-price p {
    margin: 0px;
    font-weight: 500;
}

.item-name {
    font-size: 10px;
    font-weight: bold;
}

.item-name p {
    /* margin: 0px; */
    margin-left: 10px;
}