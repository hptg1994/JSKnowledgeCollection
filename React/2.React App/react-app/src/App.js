import React from 'react';
import logo from './logo.svg';
import './App.css';
import HOOKS from './HOOKS/hooks';
import Basic from './HighOrderComponent/BasicClass';
import PictureLoader from './Components/Picture-Loader';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>Hooks</h1>
        <HOOKS />
        <h1>Basic</h1>
        <Basic />
        <h1>Picture Loader</h1>
        <PictureLoader />
      </header>
    </div>
  );
}

export default App;
