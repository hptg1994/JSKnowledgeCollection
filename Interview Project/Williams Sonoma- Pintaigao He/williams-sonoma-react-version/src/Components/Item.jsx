import React from 'react';
import "./Item.css"

let Item = (props) => {
    let img = props.item.hero.href
    let itemName = props.item.name

    let price = 0;
    if (props.item.price) {
        price = props.item.price.regular
    } else if (props.item.priceRange) {
        price = props.item.priceRange.regular.high;
    } else {
        price = 0;
    }

    return (
        <div className="item-container">
            <div className="item-container-image">
                <img src={img} alt={itemName} />
                <div className="item-container-price">
                    <p>$ {price}</p>
                </div>
            </div>
            <div className="item-name">
                <p>{itemName}</p>
            </div>
        </div>
    )
}

export default Item;

