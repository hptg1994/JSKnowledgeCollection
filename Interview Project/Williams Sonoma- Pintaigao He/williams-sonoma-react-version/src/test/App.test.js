import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme';
import MainPage from '../Pages/Main';
import DetailPage from '../Pages/Detail';
import Carousal from '../Pages/Carousal-Modal';
import data from "./Mock/data"
import Item from '../Components/Item';

let mock = jest.fn();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should render MainPage correctly', () => {
  const wrapper = shallow(<MainPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should render DetailPage correctly', () => {
  const wrapper = shallow(<DetailPage match={mock} location={{ 'state': data.groups[0] }} props={data.groups[0]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render DetailPage correctly', () => {
  const wrapper = shallow(<Carousal images={data.groups[0].images} startIndex={0} displayModal={true} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render DetailPage correctly', () => {
  const wrapper = shallow(<Item item={data.groups[0]} />);
  expect(wrapper).toMatchSnapshot();
});