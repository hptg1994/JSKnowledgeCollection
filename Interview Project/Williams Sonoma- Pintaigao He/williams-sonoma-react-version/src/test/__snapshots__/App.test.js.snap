// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DetailPage correctly 1`] = `ShallowWrapper {}`;

exports[`should render DetailPage correctly 2`] = `ShallowWrapper {}`;

exports[`should render DetailPage correctly 3`] = `ShallowWrapper {}`;

exports[`should render MainPage correctly 1`] = `ShallowWrapper {}`;
